src/main/java/core/basesyntax/report/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.report;
+
+import core.basesyntax.storage.Storage;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    @Override
+    public String getReport() {
+        StringBuilder report = new StringBuilder("fruit,quantity\n");
+        for (Map.Entry<String, Integer> entry : Storage.getFruitStorage().entrySet()) {
```

Comment:
It's a good practice to follow the encapsulation principle. The `Storage.getFruitStorage()` method exposes the internal storage details. Instead, consider providing a method in `Storage` that returns a copy of the storage data or an unmodifiable view.